packages/plugins/jwt/src/config.ts

```diff
@@ -0,0 +1,141 @@
+import { VerifyOptions } from 'jsonwebtoken';
+import { Options as JwksClientOptions } from 'jwks-rsa';
+
+type DiscriminatedUnion<K extends PropertyKey, T extends object> = {
+  [P in keyof T]: { [Q in K]: P } & T[P] extends infer U ? { [Q in keyof U]: U[Q] } : never;
+}[keyof T];
+
+type AtleastOneItem<T> = [T, ...T[]];
+
+type ExtractTokenFunction = (params: {
+  request: Request;
+  serverContext: object | undefined;
+  url: URL;
+}) => Promise<string | undefined> | string | undefined;
+
+export type TokenLookupLocation = DiscriminatedUnion<
+  'type',
+  {
+    header: {
+      name: string;
+      prefix?: string;
+    };
+    cookie: {
+      name: string;
+    };
+    fn: {
+      getToken: ExtractTokenFunction;
+    };
+  }
+>;
+
+export type JwksSource = DiscriminatedUnion<
+  'type',
+  {
+    remote: { jwksClientOptions: JwksClientOptions };
+    inline: { signingKey: string };
```

Comment:
Do you think we could also have an `environment` source that reads the signing key from the env variables ? This is most likely from where the key will be loaded if not form a JWKS.

```suggestion
    inline: { signingKey: string };
    environment: { variableName: string };
```

Reply:
As requested by @ardatan , I changed the API so this one is no longer relevant.